generate-markdown.js

```diff
@@ -1,18 +1,115 @@
 const fs = require("fs");
 
-let raw = fs.readFileSync("companies.json");
-let parsed = JSON.parse(raw);
+const getParsedJSON = filename => {
+  let rawFile = fs.readFileSync(filename);
+  let parsed = JSON.parse(rawFile);
+  return parsed;
+};
 
 const buildRow = rowArray => {
-  return `|${rowArray.join("|")}|\r`;
+  return `|${rowArray.join("|")}|\n`;
+};
+
+const buildRowsFromParsedJSON = parsedJSON => {
+  let md = "";
+
+  // header row
+  md += buildRow(["Company", "WFH", "Travel", "Visitors", "Events", "Updated", "Source"]);
+  md += buildRow(["---", "---", "---", "---", "---", "---", "---"]);
+
+  // other rows
+  for (var i = 0; i < parsedJSON.length; i++) {
+    let d = parsedJSON[i];
+    md += buildRow([d.name, d.wfh, d.travel, d.visitors, d.events, d.updated, d.sources.join(", ")]);
+  }
+
+  return md;
+};
+
+const generateMarkdownFromFile = filename => {
+  const parsed = getParsedJSON(filename);
+  const built = buildRowsFromParsedJSON(parsed);
+  return built;
 };
 
-let md = buildRow(["Company", "WFH", "Travel", "Visitors", "Events", "Source", "Updated"]);
-md += buildRow(["---", "---", "---", "---", "---", "---", "---"]);
+const generatedMarkdown = generateMarkdownFromFile("companies.json");
+
+const result = `
```

Comment:
Thanks for the great effort making this more easier to maintain!
One suggestion. Instead of generating/writing the whole markdown file, how about we only write `generatedMarkdown` part, the companies table, to a temporary file, and let user of this copy it to the MD? I feel generating the whole MD here in the script makes other parts of the page harder to update.

Reply:
My thought was to generate the existing readme file so that everything keeps working as it has been, but this can easily be broken into two steps.